bindings/bind_py_structure_manager.cc

```diff
@@ -458,6 +458,132 @@ namespace rascal {
         py::call_guard<py::gil_scoped_release>());
   }
 
+  /**
+   * Bind a getter for the feature matrix comming from BlockSparseProperty.
+   *
+   * it will return a dictionary with keys->atomic number vals->feature mat
+   * the feature matrices size is (n_centers, inner_size)
+   */
+  template <class Calculator, class ManagerCollection_t,
+            class ManagerCollectionBinder>
+  void bind_sparse_feature_matrix_getter(
+      ManagerCollectionBinder & manager_collection) {
+    manager_collection.def(
+        "get_sparse_feature_matrix",
+        [](ManagerCollection_t & managers, Calculator & calculator) {
+          using Manager_t = typename ManagerCollection_t::Manager_t;
+          using Prop_t = typename Calculator::template Property_t<Manager_t>;
+          using Keys_t = typename Prop_t::Keys_t;
+          using VecMap_t = Eigen::Map<math::Vector_t>;
+
+          auto property_name{managers.get_calculator_name(calculator, false)};
+
+          auto && property_ =
+              managers[0]->template get_property_ref<Prop_t>(property_name);
+          // assume inner_size is consistent for all managers
```

Comment:
the cases in which it would not be true correspond to a user strong arming the code in the worst possible way so I don't think it is really worth catching.

Reply:
It could also be that the underlying code changed (which will happen at some point). It is usually nicer for future developers (this can include the original author!) to document the pre-conditions of a piece of code using assertions. 

If this is a strong requirement, then an assert is better than an exception.